addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/BlankTotpActiveScanRule.java

```diff
@@ -0,0 +1,167 @@
+        return 40048;
+    }
+    @Override
+    public String getName() {
+        return "Blank code TOTP Scan Rule";
+    }
+    @Override
+    public String getDescription() {
+        return "TOTP Page found";
+    }
+    @Override
+    public int getCategory() {
+        return Category.INFO_GATHER;
+    }
+    @Override
+    public String getSolution() {
+        return "N/A";
+    }
+    @Override
+    public String getReference() {
+        return "N/A";
+    }
+    @Override
+    public void scan() {
+        try {
+            ExtensionUserManagement usersExtension =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionUserManagement.class);
+
+            // Get target URL from request
+            HttpMessage msg = getBaseMsg();
+            String targetUrl = msg.getRequestHeader().getURI().toString();
+
+            // Find session context that matches the target URL 
+            Context activeContext = null;
+            Session session = Model.getSingleton().getSession();
+            for (Context context : session.getContexts()) {
+                if (context.isInContext(targetUrl)) {
+                    activeContext = context;
+                    break;
+                }
+            }
+            BrowserBasedAuthenticationMethod browserAuthMethod = null;
+            List<AuthenticationStep> authSteps = null;
+            AuthenticationStep totpStep = null;
+            // Check if the context is found
+            if (activeContext != null) {
+                AuthenticationMethod authMethod = activeContext.getAuthenticationMethod();
+                // Check if the authentication method is browser based
+                if (authMethod instanceof BrowserBasedAuthenticationMethod) {
+                    browserAuthMethod = (BrowserBasedAuthenticationMethod) authMethod;
+                    // Check if the authentication method has TOTP step
+                    authSteps = browserAuthMethod.getAuthenticationSteps();
+                    boolean totpFound = false;
+                    for (AuthenticationStep step : authSteps) {
+                        // Checks for TOTP_field type step or currently also allows for 
+                        // custom field b/c of the way TOTP_field step currently implemented
+                        if (step.getType() == AuthenticationStep.Type.TOTP_FIELD || (step.getType() == AuthenticationStep.Type.CUSTOM_FIELD && step.getDescription().toLowerCase().contains("totp"))) {
+                            totpFound = true;
+                            totpStep = step;
+                            break;
+                        }
+                    }
+                    if (!totpFound) {
+                        return;
+                    }
+                    
+                }
+                else{
+                    //LOGGER.error("Authentication Method is not browser based.");
```

Comment:
Either actually include logging (e.g. at DEBUG level) or remove the commented code